flink-streaming-java/src/main/java/org/apache/flink/streaming/api/datastream/WindowedStream.java

```diff
@@ -800,4 +907,39 @@ public StreamExecutionEnvironment getExecutionEnvironment() {
 	public TypeInformation<T> getInputType() {
 		return input.getType();
 	}
+
+	private static <T, R, K, W extends Window> ProcessWindowFunction<T, R, K, W> wrapWindowFunction(final WindowFunction<T, R, K, W> cleanedFunction) {
```

Comment:
Yes, I know about it, but having a WrappingFunction<WindowFunction> would not be possible here because we need to create a ProcessWindowFunction. Anyway as I would get rid of the triple nesting InternalWindowFunction(ProcessWindowFunction(WindowFunction))), this is no longer an issue. By the way, I added this triple nesting because I thought you wanted to completely decouple the WindowFunction interface from runtime internals.